fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/visitors/DdlVisitor.java

```diff
@@ -401,39 +403,57 @@ private CompiledSqlFunction visitSqlInvokedFunction(@Nonnull final RelationalPar
         boolean isSqlParameterStyle = props.parameterStyle() == null || props.parameterStyle().SQL() != null;
         boolean isScalar = functionSpecCtx.returnsClause() != null &&
                 functionSpecCtx.returnsClause().returnsType().returnsTableType() == null;
-        Assert.thatUnchecked(!isScalar, "only table functions are supported");
-        Assert.thatUnchecked(isSqlParameterStyle, ErrorCode.UNSUPPORTED_OPERATION, "only sql-style parameters are supported");
-        // todo: rework Java UDFs to go through this code path as well.
-        Assert.thatUnchecked(language == InvokedRoutine.Language.SQL, ErrorCode.UNSUPPORTED_OPERATION,
-                "only sql-language functions are supported");
-        // create SQL function logical plan by visiting the function body.
-        final var parameters = getDelegate().getPlanGenerationContext().withDisabledLiteralProcessing(() ->
-                visitSqlParameterDeclarationList(functionSpecCtx.sqlParameterDeclarationList()).asNamedArguments());
-        final var sqlFunctionBuilder = CompiledSqlFunction.newBuilder()
-                .setName(functionName)
-                .addAllParameters(parameters)
-                .seal();
-        final var parametersCorrelation = sqlFunctionBuilder.getParametersCorrelation();
-        final LogicalOperator body;
-
-        // the nested fragment below serves two purposes:
-        // 1. avoid creating a top-level LSE unnecessarily.
-        // 2. add a fake quantifier with the function parameters (if any) to resolve their references in the function body
-        //    during its plan generation.
-        final var fragment = getDelegate().pushPlanFragment();
-
-        parametersCorrelation.ifPresent(quantifier -> fragment.addOperator(LogicalOperator.newUnnamedOperator(
-                Expressions.fromQuantifier(quantifier), quantifier)));
-        if (isTemporary) {
-            body = Assert.castUnchecked(visit(bodyCtx), LogicalOperator.class);
+        if (isScalar) {
+            // user defined scalar function
+            final var semanticAnalyzer = getDelegate().getSemanticAnalyzer();
+            // get parameter names and corresponding QuantifiedObjectValue
+            List<Identifier> paramNameIdList = new ArrayList<>();
+            List<QuantifiedObjectValue> paramValueList = new ArrayList<>();
+            for (RelationalParser.SqlParameterDeclarationContext sqlParameterDeclarationContext: functionSpecCtx.sqlParameterDeclarationList().sqlParameterDeclarations().sqlParameterDeclaration()) {
```

Comment:
please use `visitSqlParameterDeclarationList`, it already returns `Expressions` that have the underlying `Value` parameters.

Also, wrap please wrap it with a `DisabledLiteralProcessing` closure, we do NOT want to have the parameter constants (if any) mess with the query semantics, see below how it is done.

You can verify later that you have a single `Expression` passed as argument, and use it in the `SemanticAnalyzer` lookup call later.